app/services/transfers/service.go

```diff
@@ -126,106 +128,132 @@ func (bs *Service) SaveRecoveredTxn(txId, amount string, asset string, m memo.Me
-	isSufficient, err := bs.fees.ValidateExecutionFee(tm.Fee, tm.Amount, tm.GasPriceGwei)
+func (ts *Service) VerifyFee(tm encoding.TransferMessage) error {
+	isSufficient, err := ts.fees.ValidateExecutionFee(tm.Fee, tm.Amount, tm.GasPriceGwei)
 	if !isSufficient {
-		bs.logger.Errorf("Fee validation for TX ID [%s] failed. Provided tx reimbursement fee is invalid/insufficient. Error [%s]", tm.TransactionId, err)
-		err = bs.transactionRepository.UpdateStatusInsufficientFee(tm.TransactionId)
+		ts.logger.Errorf("Fee validation for TX ID [%s] failed. Provided tx reimbursement fee is invalid/insufficient. Error [%s]", tm.TransactionId, err)
+		err = ts.transactionRepository.UpdateStatusInsufficientFee(tm.TransactionId)
 		if err != nil {
-			bs.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusInsufficientFee, tm.TransactionId, err)
+			ts.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusInsufficientFee, tm.TransactionId, err)
 			return err
 		}
-		bs.logger.Debugf("TX with ID [%s] was updated to [%s]. Provided fee [%s]", tm.TransactionId, transaction.StatusInsufficientFee, tm.Fee)
+		ts.logger.Debugf("TX with ID [%s] was updated to [%s]. Provided fee [%s]", tm.TransactionId, transaction.StatusInsufficientFee, tm.Fee)
 		return err
 	}
 	return nil
 }
 
 // ProcessTransfer processes the transfer message by signing the required
 // authorisation signature submitting it into the required HCS Topic
-func (bs *Service) ProcessTransfer(tm encoding.TransferMessage) error {
-	authMsgHash, err := auth_message.EncodeBytesFrom(tm.TransactionId, tm.EthAddress, tm.Erc20Address, tm.Amount, tm.Fee)
+func (ts *Service) ProcessTransfer(tm encoding.TransferMessage) error {
+	gasPriceGWeiBn, err := helper.ToBigInt(tm.GasPriceGwei)
 	if err != nil {
-		bs.logger.Errorf("Failed to encode the authorisation signature for TX ID [%s]. Error: %s", tm.TransactionId, err)
 		return err
 	}
-	signatureBytes, err := bs.ethSigner.Sign(authMsgHash)
+	gasPriceWei := ethereum.GweiToWei(gasPriceGWeiBn).String()
+
+	signature, err := ts.SignAuthorizationMessage(tm.TransactionId, tm.EthAddress, tm.Erc20Address, tm.Amount, tm.Fee, gasPriceWei)
 	if err != nil {
-		bs.logger.Errorf("Failed to sign the authorisation signature for TX ID [%s]. Error: %s", tm.TransactionId, err)
+		ts.logger.Errorf("Failed to Validate and Sign TransactionID [%s]. Error [%s].", tm.TransactionId, err)
 		return err
 	}
-	signature := hex.EncodeToString(signatureBytes)
+
 	signatureMessage := encoding.NewSignatureMessage(
 		tm.TransactionId,
 		tm.EthAddress,
 		tm.Amount,
 		tm.Fee,
+		gasPriceWei,
 		signature)
 
-	sigMsgBytes, err := signatureMessage.ToBytes()
+	err = ts.PrepareAndSubmitToTopic(signatureMessage)
+	if err != nil {
+		ts.logger.Errorf("Failed to Prepare and Submit TransactionID [%s] to the HCS Topic. Error [%s].", tm.TransactionId, err)
```

Comment:
I don't think that you should log the error. All of the errors are logged inside the `PrepareAndSubmitToTopic`. Here you should only state that this operation failed.